pkg/azure/apis/validation/validation.go

```diff
@@ -200,3 +185,58 @@ func validateSecrets(secret *corev1.Secret) []error {
 	}
 	return allErrs
 }
+
+func validateImageReference(imageRef api.AzureImageReference, fldPath *field.Path) []error {
+	var allErrs []error
+
+	if isEmptyStringPtr(imageRef.URN) && isEmptyStringPtr(imageRef.CommunityGalleryImageID) && isEmptyString(imageRef.ID) {
+		return append(allErrs, field.Required(fldPath, "must specify either a image id, community gallery image id or an urn"))
+	}
+
+	if !isEmptyStringPtr(imageRef.URN) {
+		if !isEmptyStringPtr(imageRef.CommunityGalleryImageID) || !isEmptyString(imageRef.ID) {
+			return append(allErrs, field.Required(fldPath.Child("urn"), "cannot specify a urn and community gallery image id or image id in parallel"))
+		}
+
+		urnParts := strings.Split(*imageRef.URN, ":")
+		if len(urnParts) != 4 {
+			return append(allErrs, field.Required(fldPath.Child("urn"), "invalid urn format"))
+		}
+
+		for _, s := range urnParts {
+			if isEmptyString(s) {
+				allErrs = append(allErrs, field.Required(fldPath.Child("urn"), "invalid urn format, empty field"))
+			}
+		}
+
+		return allErrs
+	}
+
+	if !isEmptyStringPtr(imageRef.CommunityGalleryImageID) {
+		if !isEmptyString(imageRef.ID) {
+			return append(allErrs, field.Required(fldPath.Child("communityGalleryImageID"), "invalid community gallery image id format, empty field"))
+		}
+
+		return allErrs
+	}
```

Comment:
a unit test for this would be good.

Reply:
I agree but the public interface is `ValidateAzureSpecNSecret()` which is lacking entirely test coverage. We should fix this. For the time being we can export `validateImageReference()` for proper test coverage.